api/graphql/index.js

```diff
@@ -456,9 +463,13 @@ export function makeMutations (expressContext, userId, isAdmin, fetchOne) {
     respondToEvent: (root, { id, response }) =>
       respondToEvent(userId, id, response),
 
+    setProposalOptions: (root, { postId, options }) => setProposalOptions({ userId, postId, options }),
+
     subscribe: (root, { groupId, topicId, isSubscribing }) =>
       subscribe(userId, topicId, groupId, isSubscribing),
 
+    swapProposalVote: (root, { postId, removeOptionId, addOptionId }) => swapProposalVote({ userId, postId, removeOptionId, addOptionId }),
```

Comment:
Yeah its just a convenience mutation, to avoid having to handle making double calls on the front-end. And yes, only applicable for the single-vote situs

Reply:
but you wouldnt need a double call on the front end of addVote just removed the old one in the case of proposals that are set to only allow for one vote

Reply:
That would just keep adding votes to that proposal. Now we have to filter and sort them somewhere to determine what vote is someone's latest and we have to do that for every vote to figure out an accurate vote count. Seems like it causes a lot more hassle than it prevents

Reply:
why would it keep adding votes, you first remove the old vote, then add the new one. but im not going to push it, your way is ok too